zerver/views/auth.py

```diff
@@ -727,6 +731,10 @@ def login_page(
         # HttpResponse. See django.template.response.SimpleTemplateResponse,
         # https://github.com/django/django/blob/master/django/template/response.py#L19.
         update_login_page_context(request, template_response.context_data)
+    if 'password' in request.POST and request.user.is_authenticated \
+            and not check_password_strength(request.POST['password']):
```

Comment:
I think we should rather go with the approach of redirecting in `home_real`, so that we just refuse to even load the app and force the user to change. Perhaps we can
1. If password correct, check its strength in EmailAuthBackend.authenticate and set `.needs_to _change_password`  to True on the `UserProfile`.
2. In home_real, check `user_profile.needs_to _change_password` and if True, redirect to the password reset?
3. Clear that flag when the user changes the password.

Perhaps @timabbott can comment if this sounds reasonable.